tests/load/test_dummy_client.py

```diff
@@ -96,15 +96,15 @@ def test_unsupported_write_disposition() -> None:
     load.load_storage.normalized_packages.save_schema(load_id, schema)
```

Comment:
I needed to change a bunch of tests, since we do not rely on multiple executions of the run method anymore. All the changes make sense, it might be good to add a few more tests cases specific to the new implementation.